### PR TITLE
AKS + Microk8s - June 2021 release uplift

### PR DESCRIPTION
**Pull request by mdrakiburrahman**
Note that this contains the same June release changes to the Microk8s section as well (which will be functional once the NodePort DC images are rolled out soon).

Also, the enhancements [we discussed in the Microk8s scenario](https://github.com/microsoft/azure_arc/pull/620) has been ported over to this AKS PR.

### CODE REVIEW

File: azure_arc_data_jumpstart/aks/arm_template/artifacts/DeployPostgreSQL.ps1 [hunk 1]
```diff
@@ -42,26 +45,25 @@ Do {
...
-$pgsqlfile = "C:\Temp\postgres_instance_endpoint.txt"
-$pgsqlstring = Get-Content $pgsqlfile
-$pgsqlstring.split(" ") | Out-File "C:\Temp\postgres_instance_endpoint.txt" | Out-Null
-(Get-Content $pgsqlfile | Select-Object -Skip 7) | Set-Content $pgsqlfile
-(Get-Content $pgsqlfile | Where-Object {$_.trim() -ne "" }) | Set-Content $pgsqlfile
-$pgsqlstring = Get-Content $pgsqlfile
 
-(Get-Content -Path $settingsTemplate) -replace 'arc_postgres',$pgsqlstring | Set-Content -Path $settingsTemplate
+# Retrieving PostgreSQL connection endpoint
+$pgsqlstring = kubectl get postgresql  jumpstartps -n arc -o=jsonpath='{.status.primaryEndpoint}'
```
Comment (likamrat) [inline azure_arc_data_jumpstart/aks/arm_template/artifacts/DeployPostgreSQL.ps1:57]:
```suggestion
$pgsqlstring = kubectl get postgresql jumpstartps -n arc -o=jsonpath='{.status.primaryEndpoint}'
```

**Comment by mdrakiburrahman** [comment]
@likamrat  PR has been updated to include Microk8s scenario as well (tested with new DC image)

File: azure_arc_data_jumpstart/aks/arm_template/aks.json [hunk 2]
```diff
@@ -1,152 +1,150 @@
...
+    "osType": {
+      "type": "string",
+      "defaultValue": "Linux",
+      "allowedValues": ["Linux"],
+      "metadata": {
+        "description": "The type of operating system."
+      }
+    },
+    "kubernetesVersion": {
+      "defaultValue": "1.18.17",
```
Comment (likamrat) [inline azure_arc_data_jumpstart/aks/arm_template/aks.json:91]:
```suggestion
      "defaultValue": "1.19.11",
```

File: azure_arc_data_jumpstart/aks/arm_template/artifacts/azuredeploy.parameters.example.json [hunk 3]
```diff
@@ -1,42 +1,45 @@
...
+      "value": false
+    },
+    "deployPostgreSQL": {
+      "value": false
+    },
+    "templateBaseUrl": {
+      "value": "https://raw.githubusercontent.com/microsoft/azure_arc/main/azure_arc_data_jumpstart/aks/arm_template/"
+    },
+    "kubernetesVersion": {
+      "value": "1.18.17"
```
Comment (likamrat) [inline azure_arc_data_jumpstart/aks/arm_template/artifacts/azuredeploy.parameters.example.json:39]:
```suggestion
      "value": "1.19.11"
```

File: azure_arc_data_jumpstart/aks/arm_template/azuredeploy.parameters.json [hunk 4]
```diff
@@ -1,42 +1,45 @@
...
+      "value": "<SQL Managed Instance deployment (true/false)>"
+    },
+    "deployPostgreSQL": {
+      "value": "<PostgreSQL Hyperscale deployment (true/false)>"
+    },
+    "templateBaseUrl": {
+      "value": "https://raw.githubusercontent.com/microsoft/azure_arc/main/azure_arc_data_jumpstart/aks/arm_template/"
+    },
+    "kubernetesVersion": {
+      "value": "1.18.17"
```
Comment (likamrat) [inline azure_arc_data_jumpstart/aks/arm_template/azuredeploy.parameters.json:39]:
```suggestion
      "value": "1.19.11"
```

File: azure_arc_data_jumpstart/aks/arm_template/artifacts/Bootstrap.ps1 [hunk 5]
```diff
@@ -62,10 +64,23 @@ Disable-ieESC
...
+Invoke-WebRequest ($templateBaseUrl + "artifacts/SQLMI.parameters.json") -OutFile "C:\Temp\SQLMI.parameters.json"
+Invoke-WebRequest ($templateBaseUrl + "artifacts/postgreSQL.json") -OutFile "C:\Temp\postgreSQL.json"
+Invoke-WebRequest ($templateBaseUrl + "artifacts/postgreSQL.parameters.json") -OutFile "C:\Temp\postgreSQL.parameters.json"
+Invoke-WebRequest ($templateBaseUrl + "artifacts/wallpaper.png") -OutFile "C:\Temp\wallpaper.png"
+
 # Installing tools
 workflow ClientTools_01
         {
-            $chocolateyAppList = 'azure-cli,az.powershell,kubernetes-cli,vcredist140,microsoft-edge,azcopy10,vscode,putty.install,kubernetes-helm'
+            $chocolateyAppList = 'azure-cli,az.powershell,kubernetes-cli,vcredist140,microsoft-edge,azcopy10,vscode,putty.install,kubernetes-helm,grep'
```
Comment (likamrat) [inline azure_arc_data_jumpstart/aks/arm_template/artifacts/Bootstrap.ps1:83]: grep?

File: azure_arc_data_jumpstart/aks/arm_template/artifacts/DeployPostgreSQL.ps1 [hunk 6]
```diff
@@ -1,13 +1,15 @@
 Start-Transcript -Path C:\Temp\deployPostgreSQL.log
 
 # Deployment environment variables
-$controllerName = "Jumpstart-DC"
+$controllerName = "jumpstart-dc"
```
Comment (likamrat) [inline azure_arc_data_jumpstart/aks/arm_template/artifacts/DeployPostgreSQL.ps1:4]: This will also change the readme screenshot. need to update...

**Review by likamrat — CHANGES_REQUESTED** [review]
- AKS default version 1.18.7 > 1.19.11
- grep?
- minor screenshot update needed following _$controllerName_ value update

File: azure_arc_data_jumpstart/aks/arm_template/artifacts/Bootstrap.ps1 (hunk 5 above)
Comment (mdrakiburrahman) [reply azure_arc_data_jumpstart/aks/arm_template/artifacts/Bootstrap.ps1:83]: @likamrat - I find `grep` is useful to have in the client VM when searching for podnames etc. 

e.g. `kubectl get pods --all-namespaces | grep sql` would filter out all pods with `sql` in it, syntax is easier to use than PowerShell native `Select-String`.

It's not essential though, so happy to get it removed.

Comment (likamrat) [reply azure_arc_data_jumpstart/aks/arm_template/artifacts/Bootstrap.ps1:83]: Was just wondering, let's keep it

File: azure_arc_data_jumpstart/aks/arm_template/artifacts/DeployPostgreSQL.ps1 (hunk 6 above)
Comment (mdrakiburrahman) [reply azure_arc_data_jumpstart/aks/arm_template/artifacts/DeployPostgreSQL.ps1:4]: @likamrat no problem. Let me submit a commit with the readme updates - had created a diagram to add onto the AKS scenario.

File: azure_arc_data_jumpstart/aks/arm_template/artifacts/azuredeploy.parameters.example.json (hunk 3 above)
Comment (mdrakiburrahman) [reply azure_arc_data_jumpstart/aks/arm_template/artifacts/azuredeploy.parameters.example.json:39]: @likamrat I will test the deployment with the new AKS version once more.

**Comment by mdrakiburrahman** [comment]
@likamrat thanks. Let me test a deployment with AKS 1.19.11, and take some updated screenshots for the readme.

I will submit a commit soon with the updated readme.

File: azure_arc_data_jumpstart/aks/arm_template/artifacts/DeployPostgreSQL.ps1 (hunk 6 above)
Comment (mdrakiburrahman) [reply azure_arc_data_jumpstart/aks/arm_template/artifacts/DeployPostgreSQL.ps1:4]: Done

**Comment by mdrakiburrahman** [comment]
@likamrat:
* Tested 1.19.11 deployment with DC, MI and PG
* Updated all AKS scenario screenshots with "jumpstart-dc"
* Added 3 diagrams
* Readme text (3 scenarios) updated to include:
  * New VNET.json reference
  * --parameters templateBaseUrl
  * Diagram

**Review by likamrat — APPROVED** [review]
LGTM. Great work!

**Pull request merged**